app/controllers/Emergency.scala

```diff
@@ -31,22 +42,98 @@ object Emergency extends Controller with PanDomainAuthActions {
           Ok(views.html.emergency.reissueSuccess())
             .withCookies(authCookies: _*)
         } else {
-          unauthorised("Only Guardian email addresses with two-factor auth are supported.")
+          unauthorised("Only Guardian email addresses with two-factor auth are supported.", reissueTopic)
         }
       }
       catch {
         case e: CookieSignatureInvalidException =>
-          unauthorised("Invalid existing session, could not log you in.")
+          unauthorised("Invalid existing session, could not log you in.", reissueTopic)
         case e: CookieParseException =>
-          unauthorised("Could not refresh existing session due to a corrupted cookie.")
+          unauthorised("Could not refresh existing session due to a corrupted cookie.", reissueTopic)
       }
     }).getOrElse {
-      unauthorised("No existing login session found, unable to log you in.")
+      unauthorised("No existing login session found, unable to log you in.", reissueTopic)
+    }
+  }
+
+  def requestCookieLink = EmergencySwitchIsOnAction { req =>
+    Ok(views.html.emergency.newCookieIssued())
+  }
+
+  def sendCookieLink = EmergencySwitchIsOnAction { req =>
+
+    val tokenIssuedAt = DateTime.now().getMillis
+    val emailAddress= req.body.asFormUrlEncoded.get("email")(0)
+
+    if (emailAddress.matches("^[a-z]+\\.[a-z]+@(guardian.co.uk|theguardian.com)$")) {
```

Comment:
I think it might be worth getting the user to send firstname and last name as two separate data fields. Then we just send an email to `firstname.lastname@guardian.co.uk` and not do this check.